src/parsers.rs

```diff
@@ -709,6 +725,10 @@ fn parse_ixdtf(source: &[u8], variant: ParseVariant) -> TemporalResult<IxdtfPars
         );
     }
 
+    // TemporalTimeString has +TimeRequired
```

Comment:
I think something like this should be asserted by [`parse_time`](https://github.com/boa-dev/temporal/blob/main/src/parsers.rs#L800).